src/compile/data/fold.ts

```diff
@@ -0,0 +1,55 @@
+import {FoldTransform as VgFoldTransform} from 'vega-typings';
+import {isString} from 'vega-util';
+import {FoldTransform} from '../../transform';
+import {duplicate} from '../../util';
+import {DataFlowNode} from './dataflow';
+
+/**
+ * A class for flatten transform nodes
+ */
+export class FoldTransformNode extends DataFlowNode {
+
+  public clone() {
+    return new FoldTransformNode(this.parent, duplicate(this.transform));
+  }
+
+  constructor(parent: DataFlowNode, private transform: FoldTransform) {
+    super(parent);
+  }
+
+  public producedFields() {
+    const defaultNames = this.getDefaultName();
+
+    if (defaultNames === undefined) {
+      return {'key': true, 'value': true};
+    } else {
+      return defaultNames.reduce((result,item) => {
+        result[item] = true;
+        return result;
+      }, {});
+    }
+  }
+
+  private getDefaultName(): [string, string] | undefined {
+    const as = this.transform.as;
+    if (as && isString(as[0]) && isString(as[1])) {
```

Comment:
1) I don't think you need to check if each item is a string as that's already enforced by the schema.  Just need to check if they exist.

2) if as[0] is specified, but as[1] isn't.  Shouldn't we use `[as[0], 'value']` ?